js/bundle.go

```diff
@@ -289,6 +289,13 @@ func (b *Bundle) Instantiate(
 		jsOptionsObj = jsOptions.ToObject(rt)
 	}
 	b.Options.ForEachSpecified("json", func(key string, val interface{}) {
+		// goja.ToValue generates an empty Object `{}` for the scenarios's value.
+		// This skips the set preserving the original `options.scenarios` value.
+		//
+		// TODO: rid this part when #883 will be implemented
+		if key == "scenarios" {
+			return
```

Comment:
```suggestion
			val = (map[string]lib.ExecutorConfig)(val.(lib.ScenarioConfigs))
```

This seems like a better fix.
We basically remove the type "around" map and give goja a map which then is set correctly.

This definitely needs a bunch of tests though

Reply:
Probably a better variant will be for this to have some kind of interface `ToValue` or something like that. Then we can check that `val` implements it and call the method to get the actual value - the body in this will be this same line. 

But I feel that this might be too much work especially as this will likely be the only "exception" so maybe leave that as a  comment and hope that we will never have to implement it :crossed_fingers:

Reply:
Unfortunately, this doesn't work really well because it includes the BaseConfig that the executor's config contains. The ideal approach would be to use a mapper like ToValue to return the correct object.